pkg/machines/libvirt.es6

```diff
@@ -150,7 +150,7 @@ LIBVIRT_PROVIDER = {
      * @param VM name
      * @returns {Function}
      */
-    GET_VM ({ lookupId: name, connectionName }) {
+    GET_VM ({ name, connectionName }) {
```

Comment:
Perhaps this file should now be renamed to libvirt-virsh.es6, for symmetry with libvirt-dbus.es6 and to avoid confusion?

Also, everything here is still name based, whereas -dbus uses Id, and the provider-actions get both. This is a bit confusing as well, and the provider action could potentially get inconsistent name and ID. It  might be clearer to just declare the name being the "id" here, and rename everything "id"? It should be treated as an opaque, backend-defined value, so the UI should not care about it and just pass it around, and the actions would then only have one ID.

Reply:
libvirt.es6 renaming to libvirt-virsh.es6 done.

Regarding the id comment I have some worries.

* In case we are calling the actions from the react component, we should not deal with what is the underlying implementation. 
See here:
https://github.com/cockpit-project/cockpit/blob/4edbf97298bc85fb4aa17196b52cbd110dc1e79c/pkg/machines/components/diskAdd.jsx#L424
If we follow the pattern you suggest, and call it like getVm({id: vm.name, connectionName}) this will work only for the libvirt-virsh and with id it will work only for dbus provider. It's of course possible to find out what provider is used, but it's not correct to check it there I think.
 * In libvirt-dbus provider methods, we need mostly both name and id attributes in the parameters. Id to use it the API calls and name to print errors to the user if something went bad. User probably wants to see name in the error message and not VM id.

So, considering these I think it makes sense to keep both attributes passed to each action, even though sometimes they are not used. Also, I am trying to use everywhere named args, so that order of parameters will not matter.

Reply:
I see, thanks for the explanation! Makes sense. (Good to discuss this stuff and have a written record of it :smile: )